src/tests/ftest/container/open.py

```diff
@@ -53,6 +51,22 @@ class OpenContainerTest(TestWithServers):
 
     :avocado: recursive
     """
+    def __init__(self, *args, **kwargs):
+        """Initialize member."""
+        super(OpenContainerTest, self).__init__(*args, **kwargs)
+        self.expected_result = RESULT_PASS
+
+    def tearDown(self):
+        # If we call container.open with invalid parameter, container destroy
+        # during the tearDown causes an error. In order to prevent it, set None
+        # to self.container so that container destroy won't happen and it'll
+        # proceed to pool destroy. On the other hand, if we open container with
+        # valid parameter, we need to destroy it during the tearDown; if we
+        # set None, it'll skip container destroy step, try to destroy pool, and
+        # cause an error because the container in the pool isn't destroyed.
+        if self.expected_result == RESULT_FAIL:
+            self.container = None
+        super(OpenContainerTest, self).tearDown()
```

Comment:
I don't believe this accounts for the case where the test expect the container open to fail, but it incorrectly succeeds - and therefore needs to be destroyed.   Consider setting `self.container = None` when the test detects that a test was not created successfully - regardless of the expected result.

The other thing that may be worth looking into is `TestContainer` has `opened` and `attached` attributes that are used by `destroy_containers` (https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/apricot/apricot/test.py#L370) to determine if the container should be closed and destoyed.  That code should handle this case, so why might those attributes not be being set correctly?

Reply:
Thanks for the suggestion. I found out that if we use API, container destroy fails for some reason. I tried daos command and it works, so modified the code to use daos. There were some issues related to daos like attached = 1 not being set, so I fixed them.